cmd/security-file-token-provider/main.go

```diff
@@ -0,0 +1,126 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap"
+	bootstrapinterface "github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/interfaces"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/startup"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/usage"
+	"github.com/edgexfoundry/go-mod-registry/registry"
+
+	"github.com/edgexfoundry/edgex-go/internal/security/authtokenreader"
+	"github.com/edgexfoundry/edgex-go/internal/security/fileioperformer"
+	"github.com/edgexfoundry/edgex-go/internal/security/fileprovider"
+	"github.com/edgexfoundry/edgex-go/internal/security/tokenconfig"
+	"github.com/edgexfoundry/edgex-go/internal/security/vaultclient"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+// Constants
+
+// Dependencies
+
+var fileProvider fileprovider.TokenProvider
+
+// Flag variables
+
+var configDir string
+var profileDir string
+var useRegistry bool
+
+// define and register command line flags
+func init() {
```

Comment:
Is this called from anywhere?

Reply:
Content should be moved into main (following pattern of core services) along with corresponding flag variables.  Seems like you'd also want to update config-seed (maybe you already have in a separate PR?) to push the relevant configuration.toml configuration to registry for retrieval/usage by this app if/when --registry/-r flag is set.

Reply:
> Is this called from anywhere?

Yes, init() is a [package initialization hook](https://golang.org/ref/spec#Package_initialization) and is called by golang.

> Content should be moved into main (following pattern of core services) along with corresponding flag variables.

It is actually necessary to have it as moving it to `main()` will break the unit tests because calling flag.XXX functions multiple times will result in a runtime error.

> Seems like you'd also want to update config-seed (maybe you already have in a separate PR?) to push the relevant configuration.toml configuration to registry for retrieval/usage by this app if/when --registry/-r flag is set.

Will look into it.

Reply:
I'd still counsel you to move the code.

Your tests are acceptance tests and not unit tests.  I wonder if that's really what you want here.  I'm not sure I see the point in exercising the common bootstrap package in your tests (or taking the dependencies related to it).

You might get better results (and be able to test more variation) if you separate out the content of your bootstrap handler into its own function and use that as an entry point for your tests.

Reply:
Modified as suggested and rebased.  Please re-review.